alica_engine/src/engine/RunningPlan.cpp

```diff
@@ -480,7 +480,7 @@ void RunningPlan::deactivate()
     deactivateChildren();
 
     if (isBehaviour()) {
-        _ae->editBehaviourPool().stopBehaviour(*this);
+        _basicBehaviour->stop();
```

Comment:
I think we should also change how RunningPlan's are destroyed. Right now, a vector of RunningPlan shared_ptrs are stored in the PlanBase which is iterated over to find RunningPlan's that can be deleted by calling RunningPlan::isDeletable. One reason why this was required earlier is because the behaviours attached to a running plan could still be running even if we transitioned out of it, but with this refactor this won't be the case. isDeletable does check a few more things but I think we should be able to destroy the RunningPlan when it is deactivated.

Reply:
Yes, would do it in the 3rd cleanup pr